master/internal/kubernetes/pods.go

```diff
@@ -342,6 +345,14 @@ func (p *pods) handleGetAgentsRequest(ctx *actor.Context) {
 func (p *pods) summarize(ctx *actor.Context) map[string]agent.AgentSummary {
 	podHandlers := make([]*actor.Ref, 0, len(p.podNameToPodHandler))
 	for _, podHandler := range p.podNameToPodHandler {
+		// Skip pod actors that have not yet received a pod update
```

Comment:
blocking: The current `pod` actor process messages that might take various lengths of time, which probably is not very ideal. One solution is to use two actors, one processing quick requests, such as status updating and summarizing, and the other processing slow requests, such as creation and termination. That is also the reason I think putting creating a separate actor rather letting the pod actors be stuck in `PreStart` message would be helpful.